rb/spec/integration/selenium/webdriver/bidi_spec.rb

```diff
@@ -52,6 +57,32 @@ module WebDriver
           level: BiDi::LogInspector::LOG_LEVEL[:ERROR]
         )
       end
+
+      it 'does not close BiDi session if one window is closed' do
+        status = driver.bidi.session.status
+        expect(status.ready).to be false
+        expect(status.message).to be_a String
+
+        driver.switch_to.new_window(:window)
+        driver.switch_to.new_window(:tab)
+        driver.switch_to.new_window(:tab)
+
+        driver.close
+
+        status_after_closing = driver.bidi.session.status
+        expect(status_after_closing.ready).to be false
+        expect(status_after_closing.message).to be_a String
+      end
+
+      it 'closes BiDi session if last window is closed' do
+        status = driver.bidi.session.status
```

Comment:
Optional. Could maybe isolate all these status calls to a let block at the top.

Reply:
The real question is whether session should be memoized here: https://github.com/SeleniumHQ/selenium/blob/5ce55558cbadcbc6d4172dc51be98bf21287c1e4/rb/lib/selenium/webdriver/bidi.rb#L39
@p0deje is there a reason we create a new `Session` instance for each call?

Reply:
Hm, I suppose we can memoize it, why not?

Reply:
@TamsilAmani So, let's change line 40 there to `@session ||= Session.new(self)`
If that's memoized, then there's no advantage to using a `let` statement in the specs, multiple calls execute the desired method on the same instance.

Reply:
Changes work as expected. Thanks!